src/block_cache/block_tier.c

```diff
@@ -146,13 +156,16 @@ __wt_blkcache_get_handle(WT_SESSION_IMPL *session, WT_BM *bm, uint32_t objectid,
     __wt_writelock(session, &bm->handle_array_lock);
 
     /* Check to see if the object was added while we opened it. */
-    __blkcache_find_open_handle(bm, objectid, blockp);
+    __blkcache_find_open_handle(bm, objectid, reading, blockp);
 
     if (*blockp == NULL) {
```

Comment:
Not a part of this ticket, but what happens when `*blockp` is not `NULL`? Do we assume that it would be equal to `new_handle`? If so, it would be useful to state that explicitly through an assertion (it would also help with the code readability).

Reply:
Good idea. Done! (And yes, we should get back `new_handle` if the object is in the handle list)